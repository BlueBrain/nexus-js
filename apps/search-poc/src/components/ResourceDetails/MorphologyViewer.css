
.morphology {
  height: 400px;
}
