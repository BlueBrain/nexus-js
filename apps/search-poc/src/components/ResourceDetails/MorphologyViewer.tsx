import * as React from 'react';
import { Card, Spin, Empty } from 'antd';
import './MorphologyViewer.css';

const MorphologyViewer: React.FunctionComponent<{
  loading: boolean;
  error: Error | null;
  stage?: string;
  morphology: any;
  ref: React.RefObject<HTMLDivElement>;
}> = ({ loading, error, ref, stage }) => {
  return (
    <Spin spinning={loading} tip={stage}>
      <Card hoverable className="morphology-viewer">
        <div className={`viewer ${!loading && !error && '-loaded'}`} ref={ref}>
          {error && <Empty>{error.message}</Empty>}
        </div>
      </Card>
    </Spin>
  );
};

export default MorphologyViewer;
