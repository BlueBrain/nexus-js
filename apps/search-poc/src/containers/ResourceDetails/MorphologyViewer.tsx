import * as React from 'react';
import { Resource } from '@bbp/nexus-sdk';

import useMorphology from '../../hooks/useMorphology';
import { parseProjectUrl } from '../../utils';
import { MINDSResource } from './types';
import MorphologyViewer from '../../components/ResourceDetails/MorphologyViewer';
import { get } from 'lodash';

const MorpologyViewerContainer: React.FunctionComponent<{
  resource: Resource & MINDSResource;
}> = props => {
  const id = get(props, 'resource.distribution.contentUrl');
  const ref = React.useRef<HTMLDivElement>(null);
  const [org, proj] = parseProjectUrl(props.resource._project);
  const { morphology, loading, stage, error } = useMorphology(
    {
      org,
      proj,
      id,
    },
    ref,
  );

  return MorphologyViewer({
    morphology,
    loading,
    stage,
    error,
    ref,
  });
};

export default MorpologyViewerContainer;
