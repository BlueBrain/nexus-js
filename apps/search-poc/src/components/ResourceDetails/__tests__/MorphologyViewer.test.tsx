import MorphologyViewer  from '../MorphologyViewer';
import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';


it('MorphologyViewer renders correctly', () => {
    const divElement = React.createRef<HTMLDivElement>();
    const props = {
        loading: true,
        error: null,
        stage: 'stage',
        morphology: 'morphology',
        ref: divElement,
      };


    const result = shallow(<MorphologyViewer {...props} />);
    expect(result).toMatchSnapshot();
});