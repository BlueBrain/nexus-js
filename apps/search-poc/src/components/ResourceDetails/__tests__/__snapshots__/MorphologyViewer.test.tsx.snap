// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MorphologyViewer renders correctly 1`] = `
<Spin
  size="default"
  spinning={true}
  tip="stage"
  wrapperClassName=""
>
  <Card
    className="morphology-viewer"
    hoverable={true}
  >
    <div
      className="viewer false"
    />
  </Card>
</Spin>
`;
